### PR TITLE
Last particle pair amplitude fixed to exactly amp(1), do not caluculate.

### PR DESCRIPTION
**Pull request by kyrsjo**

### CODE REVIEW

**Comment by kyrsjo** [comment]
Two points:
- If we do this, then let's also use crlibm for the I/O for fort.12 and fort.13. This will make testing quite a bit easier.
- This should be version 4.7.0.

**Comment by rdemaria** [comment]
Which test did fail after all? I agree for crlibm in fort.12/13...

**Comment by kyrsjo** [comment]
I can't remember which tests failed, if any. Will run tests tomorrow, note what breaks, and then fix fort.12/13 and run tests again.

File: SixTrack/sixtrack.s [hunk 1]
```diff
@@ -24549,7 +24549,12 @@ C Should get me a NaN
             bet0v(ib2,2)=bet0(2)
             alf0v(ib2,1)=alf0(1)
             alf0v(ib2,2)=alf0(2)
-            ampv(ib2)=amp(1)-damp*dble(ib1-1)                            !hr05
+            ampv(ib2)=amp(1)-damp*dble(ib1-1) !hr05
+            if (ib1.eq.napx) then
```
Comment (rdemaria) [inline SixTrack/sixtrack.s]: if (ib1.eq.(napx-1)) then
                ampv(ib2)=amp0

Appear to work

**Comment by kyrsjo** [comment]
Can you post the example?

**Comment by kyrsjo** [comment]
Tests now failing (from standard build):
 * dynk_globalvars
 * javier
 * dump_binary
 * beam-HO_4D-newstyle
 * beam-HO_4D-oldstyle
 * beam-HO_6D-newstyle
 * beam-HO_6D-oldstyle
 * beam-HO_LR-newstyle
 * beam-HO_LR-oldstyle
 * beam-HO_LR-ibbc-oldstyle
 * beam-HO_LR-ibbc-newstyle

Nothing in medium was failing, however prob1/prob3 also have 30 pairs so I expect them to fail as well.

**Comment by kyrsjo** [comment]
Hmm, there seems to be some issues when there is a single pair also, and the first- and last amplitudes are not the same. *investigating*

**Comment by rdemaria** [comment]
Probably it is what you meant, but strictly is the  first and last pair
that should be the same (so first with second last and second with last)

**Comment by kyrsjo** [comment]
This problem only occurs when there is a single amplitude; it will then now always use the highest edge of the range - before it would then always use the lowest edge of the range.

It's an easy fix.

**Comment by kyrsjo** [comment]
Now all fast/medium tests are passing... Adding Riccardo's test which should break it.

**Comment by kyrsjo** [comment]
It broke it - from singletrackfile.dat, last pair on the first turn:
```
 Read record           59
 Turn            0    Particle           59
  0.0000000000000000e+000 
  4.0768691637944503e+000 
 -6.7321483365972365e-002 
  1.0970556168657095e+000 
  1.3634257115435790e-002 
 -6.0045217724348114e-002 
  7.5465078169200805e-004 
  450000.00000000000e+000 
 Turn            0    Particle           60
  0.0000000000000000e+000 
  4.0768701637944504e+000 
 -6.7321483365972365e-002 
  1.0970556168657095e+000 
  1.3634257115435790e-002 
 -6.0045217724348114e-002 
  7.5465078169200805e-004 
  450000.00000000000e+000 
```
Previously:
```
 Read record           59
 Turn            0    Particle           59
  0.0000000000000000e+000 
  4.0768691637944503e+000 
 -6.7321483365972351e-002 
  1.0970556168657091e+000 
  1.3634257115435786e-002 
 -6.0045217724348107e-002 
  7.5465078169200805e-004 
  450000.00000000000e+000 
 Turn            0    Particle           60
  0.0000000000000000e+000 
  4.0768701637944504e+000 
 -6.7321483365972351e-002 
  1.0970556168657091e+000 
  1.3634257115435786e-002 
 -6.0045217724348107e-002 
  7.5465078169200805e-004 
  450000.00000000000e+000 
```

**Comment by kyrsjo** [comment]
Using the "b" version of the test provided by Riccardo, which has TRACK block
```
TRACKING PARAMETERS-----------------------------------------------------
100 0 30 3.79111 5.68667 0 1
1 1 0 1 2
0 0 1 1 1 50000 2
NEXT
```
VS in the "a" version (the current `last_amp` test)
```
TRACKING PARAMETERS-----------------------------------------------------
100 0 30 1.89556 3.79111 0 1
1 1 0 1 2
0 0 1 1 1 50000 2
NEXT
```
the last particle in `a` -- extracted from singletrackfile.dat using
`./read90 --STF --fname singletrackfile.dat --ofname firstparticle.dat --SP 1`
is identical to the first particle in `b`, which was extracted using
`./read90 --STF --fname singletrackfile.dat --ofname lastparticle.dat --SP 59`.

I'll update the docs, and then let's bump the version number (last digit) and merge this.

**Pull request merged**